File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Examples2D/CurvedCNS2D/BoxFlowIC2D.cpp
// BoxFlowIC2D.m
// function [rho, rhou, rhov, Ener] = BoxFlowIC2D(x, y, time)
// 2007/06/21
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "CurvedCNS2D.h"


//---------------------------------------------------------
void CurvedCNS2D::BoxFlowIC2D
(
  const DVec&   xi,   // [in]
  const DVec&   yi,   // [in]
        double  ti,   // [in]
        DMat&   Qo    // [out]
)
//---------------------------------------------------------
{
  // function Q = BoxFlowIC2D(x, y, time)
  //
  // Purpose: compute plane flow configuration 

  //-------------------------------------
  // adjust parameters
  //-------------------------------------
  this->gamma = 1.4;
  this->gm1   = 0.4;  // (gamma-1)

  //-------------------------------------
  // use wrappers to update Qo in-place
  //-------------------------------------
  DVec rho,rhou,rhov,Ener; int Nr=Np*K;
  rho.borrow (Nr, Qo.pCol(1));
  rhou.borrow(Nr, Qo.pCol(2));
  rhov.borrow(Nr, Qo.pCol(3));
  Ener.borrow(Nr, Qo.pCol(4));

  if (1) 
  {
    this->pref = 12.0;

    rho  =  1.0;
    rhou = -sin(2.0*pi*y);
    rhov =  sin(4.0*pi*x);
    Ener = pref/gm1 + 0.5*(sqr(rhou) + sqr(rhov)).dd(rho);
  } 
  else 
  {
    this->pref = 12.0;

    rho  = 1.0;
    rhou = 0.0;
    rhov = 0.0;
    Ener = pref/gm1 + 0.5 * rho.dm(exp(-4.0*(sqr(cos(pi*x))+sqr(cos(pi*y)))));
  }
}

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/CFD2D/BoxFlowIC2D.m
function [rho, rhou, rhov, Ener] = BoxFlowIC2D(x, y, time)
  
% function Q = BoxFlowIC2D(x, y, time)
% Purpose: compute plane flow configuration 

gamma = 1.4; pbar = 10;
Np = size(x,1); K = size(x,2);

if(1)

  pref = 12;
  rho = ones(Np, K);
  rhou = -sin(2*pi*y);
  rhov =  sin(4*pi*x);
  Ener = pref/(gamma-1) + 0.5*(rhou.^2 + rhov.^2)./rho;
else

  pref = 12;
  rho  = ones(Np,K);
  rhou = zeros(Np,K);
  rhov = zeros(Np,K);
  Ener = pref/(gamma-1) + 0.5*rho.*exp(-4*(cos(pi*x).^2 + cos(pi*y).^2));
end
return;
